opensearch/src/main/java/org/opensearch/sql/opensearch/storage/scan/OpenSearchIndexScan.java

```diff
@@ -45,14 +45,18 @@ public class OpenSearchIndexScan extends TableScanOperator implements Serializab
   /** Search response for current batch. */
   private Iterator<ExprValue> iterator;
 
-  /** Creates index scan based on a provided OpenSearchRequestBuilder. */
   public OpenSearchIndexScan(
       OpenSearchClient client, int maxResponseSize, OpenSearchRequest request) {
-    this.client = client;
     this.maxResponseSize = maxResponseSize;
+    this.client = client;
     this.request = request;
   }
 
+  /** Creates index scan based on a provided OpenSearchRequestBuilder. */
+  public OpenSearchIndexScan(OpenSearchClient client, OpenSearchRequest request) {
+    this(client, -1, request);
```

Comment:
just change `private int maxResponseSize;` to `private int maxResponseSize = Int.Max`; than we can delete setting `-1` and special check `maxResponseSize > 0`.